src/main/java/cz/cvut/kbss/termit/service/business/UserService.java

```diff
@@ -215,7 +258,7 @@ public void unlock(UserAccount account, String newPassword) {
     }
 
     private void ensureNotOwnAccount(UserAccount account, String operation) {
-        if (securityUtils.getCurrentUser().equals(account)) {
+        if (securityUtils.isAuthenticated() && securityUtils.getCurrentUser().equals(account)) {
```

Comment:
So that I can reuse the logic to unlock the user account when resetting the password.
There is no authenticated user at that time.

Reply:
I don't understand. `securityUtils.getCurrentUser()` will not return user if they are not authenticated anyway, so the `securityUtils.isAuthenticated` seems unnecessary.

Reply:
Or is it to prevent an exception from being thrown?

Reply:
> Or is it to prevent an exception from being thrown?

NullPointerException, yes  
or I could change the equation to `account.equals(securityUtils.getCurrentUser())`

Reply:
Would that help, though? If you look at the `SecurityUtils.getCurrentUser` code, the NPX would probably be thrown there (or maybe ClassCastException). Anyway, I now see the reasoning. Let's keep this as is.